src/Microsoft.AspNetCore.Mvc.Core/DependencyInjection/MvcCoreServiceCollectionExtensions.cs

```diff
@@ -81,12 +81,14 @@ private static ApplicationPartManager GetApplicationPartManager(IServiceCollecti
                 manager = new ApplicationPartManager();
 
                 var environment = GetServiceFromCollection<IHostingEnvironment>(services);
-                if (string.IsNullOrEmpty(environment?.ApplicationName))
+                var entryAssemblyName = environment?.ApplicationName;
+                if (string.IsNullOrEmpty(entryAssemblyName))
                 {
                     return manager;
                 }
 
-                var parts = DefaultAssemblyPartDiscoveryProvider.DiscoverAssemblyParts(environment.ApplicationName);
+                var parts = DefaultAssemblyPartDiscoveryProvider.DiscoverAssemblyParts(entryAssemblyName)
+                    .OrderBy(part => string.Equals(entryAssemblyName, part.Name, StringComparison.Ordinal) ? 0 : 1);
```

Comment:
I think we need a secondary sort here so that it will be stable. Should we add a `.ThenBy(...)` sorted on name?

Reply:
Sure